### DISCUSSION TITLE
Where to add textual[dev] at pyproject.toml? (poetry)

### DISCUSSION THREAD

**Discussion by sauljabin** [Textualize/textual#1953] [Q&A] [answered]
Hello guys quick question, here https://textual.textualize.io/getting_started/#installation says: run `pip install "textual[dev]"`.

I'm using poetry, where (or how) should I add it?

example of poetry:
```
[tool.poetry.dependencies]
python = "^3.10"

[tool.poetry.dev-dependencies]
pytest = "^5.2"
```

**Comment by TomJGooding** [comment, marked as answer]
I don't personally use Poetry but if you're writing your `pyproject.toml` file by hand, I think you would add this to your development dependencies like this:

```
[tool.poetry.dependencies]
python = "^3.10"
textual = ">=0.11.0"

[tool.poetry.group.dev.dependencies]
pytest = "^5.2"
textual = {extras = ["dev"], version = ">=0.11.0"}
```

It looks like [the proper way to define a dev dependencies group](https://python-poetry.org/docs/master/managing-dependencies/) since Poetry 1.2.0 is `[tool.poetry.group.dev.dependencies]` - it warns that the `dev-dependencies` notation will soon be deprecated.

P.S. You might want to specify a newer version of pytest as version 5.2 was released back in 2019. The [caret requirements](https://python-poetry.org/docs/dependency-specification#caret-requirements) would not allow any versions 6.0.0 or newer.

**Reply by sauljabin** [reply]
Thanks!, really helpful.
